### PR TITLE
UM-025 - Vote for Multiple Maps

### PR DESCRIPTION
**Pull request by UrsulaMejor**
## About the PR 

1) You can now vote for multiple maps. Each map you vote for has equal weighting

2) Preferred map now counts for one (1) whole vote instead of 0.25 of a vote

## Why's this needed? 

1) First Past The Post with Single Ballot sucks

2) Old Behavior isn't intuitive, and I want to see how this plays out. Passive votes are now "worth less" because active votes can be for multiple maps. If passive votes at full power turn out to be a bad idea, maybe we just shouldn't have passive votes.

## Changelog

```
(u)UrsulaMajor:
(*)You can now vote for multiple maps
```

### CODE REVIEW

**Comment by Flaborized** [comment]
yes this i s agood idea

File: code/mapSwitching.dm [hunk 1]
```diff
@@ -352,59 +352,14 @@ var/global/datum/mapSwitchHandler/mapSwitcher
...
-			if (mapName in report_output)
-				report_output[mapName] += ckey
-			else
-				report_output[mapName] = list(ckey)
-
 	//rudely cancel the vote without counting votes/doing anything
 	proc/cancelMapVote()
 		src.playersVoting = 0
-		src.playerVotes = new()
+		//src.playerVotes = new()
```
Comment (ZeWaka) [inline code/mapSwitching.dm:358]:
```suggestion
```

File: code/mob_stat.dm [hunk 2]
```diff
@@ -148,11 +148,15 @@ var/global/datum/mob_stat_thinker/mobStat = new
 		if (world.time - mobStat.last_update > mobStat.update_interval)
 			mobStat.update()
 
+		/*
```
Comment (ZeWaka) [inline code/mob_stat.dm:151]: wait so are you deleting the map vote time in the stat panel? this is commented out but the saveStat above is not.

File: code/mapSwitching.dm [hunk 3]
```diff
@@ -486,8 +441,149 @@ var/global/datum/mapSwitchHandler/mapSwitcher
...
+			client_vote_map[map_name] = 0
+
+	proc/special_vote(var/client/C,var/map_name)
+		if(!(C.ckey in vote_map))
+			setup_client_vote_map(C)
+		var/list/client_vote_map = vote_map[C.ckey]
+		client_vote_map[map_name] = 1
+
+	proc/voting_box(var/obj/voting_box/V,var/map_name)
+		var/vref = "\ref[src]"
```
Comment (ZeWaka) [inline code/mapSwitching.dm:548]:
```suggestion
		var/vref = "\ref[V]"
```

I would probably just use `var/vref = V` here though?

Comment (UrsulaMejor) [reply code/mapSwitching.dm:548]: resolved in pending commit

File: code/mapSwitching.dm (hunk 1 above)
Comment (UrsulaMejor) [reply code/mapSwitching.dm:358]: resolved in pending commit

File: code/mob_stat.dm (hunk 2 above)
Comment (UrsulaMejor) [reply code/mob_stat.dm:151]: No, just the part where it shows your your current vote (since I don't want to built the list every tick)

**Pull request merged**